app/reducers/utils/getXDomain.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Find the min and max x values from a series within scatter/bubble chart data
+ *
+ * @param {Array} values Series values
+ * @return {Array} [min, max]
+ */
+export function getSeriesMinMax(values) {
+  return values.reduce((acc, { x }, idx) => {
+    if (0 === idx) {
+      return [x, x];
+    }
+    const newAcc = [...acc];
+    if (x < newAcc[0]) {
+      newAcc[0] = x;
+    } else if (x > newAcc[1]) {
+      newAcc[1] = x;
+    }
+    return newAcc;
+  }, [0, 0]);
+}
+
+/**
+ * Find the min and max x values from scatter/bubble chart data
+ *
+ * @param {Array} data Chart data
+ * @return {Array} [min, max]
+ */
+export default function getXDomain(data) {
```

Comment:
Any reason, you decided to put this logic here, instead of the chartTypeConfigs, which hold the custom yDomain functions?